comid/extensions.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 Contributors to the Veraison project.
```

Comment:
I understand the overall (highly valuable) goal here, but it's not entirely clear to me what the criterion used for scoping the various "constrainers" types is.  I.e., are these all and only constrainers we are going to need?

(I guess there are going to be CoRIM constrainers as well, e.g., to check that a given CoRIM only contains CoMID tags, etc.)

Reply:
I.e., are these all and only constrainers we are going to need?

There is a constrainer interface corresponding to each map-extension point.  These are the only constrainers we can have, since contraints imposed by extensions can/should only be applied to the structures being extended.

(I guess there are going to be CoRIM constrainers as well, e.g., to check that a given CoRIM only contains CoMID tags, etc.)

There is already a constrainer for that in the `corim/corim` subpackage:

https://github.com/veraison/corim/blob/feature-extensions/corim/extensions.go#L13